mule-extensions-api/src/main/java/org/mule/runtime/extension/api/dsl/syntax/XmlDslSyntaxResolver.java

```diff
@@ -406,54 +410,71 @@ public void visit(SourceModel sourceModel) {
 
       @Override
       public void visit(ConstructModel model) {
-        model.getNestedComponents().forEach(child -> child.accept(new NestableElementModelVisitor() {
+        processComposableModel(model, dsl);
+      }
 
-          @Override
-          public void visit(NestedComponentModel component) {
-            // no-op
-          }
+      @Override
+      public void visit(NestedComponentModel model) {
+        processComposableModel(model, dsl);
+      }
 
-          @Override
-          public void visit(NestedChainModel component) {
-            // no-op
-          }
+      @Override
+      public void visit(NestedChainModel model) {
+        processComposableModel(model, dsl);
+      }
 
-          @Override
-          public void visit(NestedRouteModel component) {
-            dsl.containing(component.getName(), resolveRouteDsl(component));
-          }
-        }));
+      @Override
+      public void visit(NestedRouteModel model) {
+        processComposableModel(model, dsl);
       }
+    });
+  }
+
+  private void processComposableModel(ComposableModel model, DslElementSyntaxBuilder dsl) {
+    model.getNestedComponents().forEach(child -> child.accept(new ComponentModelVisitor() {
 
       @Override
-      public void visit(NestedComponentModel model) {
+      public void visit(OperationModel model) {
```

Comment:
why are these no-op?

Reply:
I supposed that there were no cases where, a ComponentModel has as nested components Operations/Sources/Constructs. Is this assumption wrong? I am now thinking that I should make the same process as in the other cases.